File: TrackIt.Model/Helper/Meta.cs
namespace TrackIt.Model.Helper
{
	public class Meta
	{
		public int? Count { get; set; }
		public int CurrentPage { get; }
		public int TotalPages { get; }
		public bool HasPrevious => CurrentPage > 0;
		public bool HasNext => CurrentPage < TotalPages - 1;
		public Meta(int count, int pageNumber, int pageSize)
		{
			CurrentPage = pageNumber;
			Count = count;
			TotalPages = (int)Math.Ceiling(count / (double)pageSize);
		}
	}
}

File: UI/trackit_mobile/lib/models/Meta/meta.dart
import 'package:json_annotation/json_annotation.dart';
part 'meta.g.dart';

@JsonSerializable()
class Meta {
  int count = 0;
  int currentPage = 0;
  int totalPages = 0;
  bool hasPrevious = false;
  bool hasNext = false;

  Meta(this.count, this.currentPage, this.totalPages, this.hasPrevious,
      this.hasNext);

  factory Meta.fromJson(Map<String, dynamic> json) => _$MetaFromJson(json);

  Map<String, dynamic> toJson() => _$MetaToJson(this);
}
